spec.html

```diff
@@ -42207,11 +42207,19 @@ <h1>
           1. Assert: _W_ is equivalent to AgentSignifier().
           1. Assert: _W_ is on the list of waiters in _WL_.
           1. Assert: AgentCanSuspend() is *true*.
-          1. Perform LeaveCriticalSection(_WL_) and suspend _W_ for up to _timeout_ milliseconds, performing the combined operation in such a way that a notification that arrives after the critical section is exited but before the suspension takes effect is not lost. _W_ can notify either because the timeout expired or because it was notified explicitly by another agent calling NotifyWaiter with arguments _WL_ and _W_, and not for any other reasons at all.
+          1. If _minimumTimeout_ is +∞, then
+            1. Perform LeaveCriticalSection(_WL_) and suspend _W_, performing the combined operation in such a way that a notification that arrives after the critical section is exited but before the suspension takes effect is not lost. _W_ can wake from suspension because it was notified explicitly by another agent calling NotifyWaiter with arguments _WL_ and _W_, and not for any other reasons at all.
+          1. Else,
+            1. Let _additionalTimeout_ be an implementation-defined non-negative mathematical value.
+            1. Let _timeout_ be _minimumTimeout_ + _additionalTimeout_.
+            1. Perform LeaveCriticalSection(_WL_) and suspend _W_ for up to _timeout_ milliseconds, performing the combined operation in such a way that a notification that arrives after the critical section is exited but before the suspension takes effect is not lost. _W_ can wake from suspension either because the timeout expired or because it was notified explicitly by another agent calling NotifyWaiter with arguments _WL_ and _W_, and not for any other reasons at all.
```

Comment:
I think the core step is intricate enough that it shouldn't be repeated.
```suggestion
          1. Let _additionalTimeout_ be an implementation-defined non-negative mathematical value.
          1. Let _timeout_ be _minimumTimeout_ + _additionalTimeout_.
          1. NOTE: If _minimumTimeout_ is +∞, then _timeout_ is also +∞ and the following step can terminate only by another agent calling NotifyWaiter.
          1. Perform LeaveCriticalSection(_WL_) and suspend _W_ for up to _timeout_ milliseconds, performing the combined operation in such a way that a notification arriving after the critical section is exited but before the suspension takes effect is not lost. _W_ can wake from suspension either because the timeout expired or because it was notified explicitly by another agent calling NotifyWaiter with arguments _WL_ and _W_, and not for any other reasons at all.
```

Reply:
Fair enough. We chatted about this in the editor call and @bakkot agrees. I've re-merged the 2 branches.